src/js/components/PodView.js

```diff
@@ -0,0 +1,370 @@
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'status',
+        render: this.renderColumn_Status,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'cpus',
+        render: this.renderColumn_Resource,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'mem',
+        render: this.renderColumn_Resource,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'updated',
+        render: this.renderColumn_Updated,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'version',
+        render: this.renderColumn_Version,
+        sortable: true
+      }
+    ];
+  }
+
+  getTableDataFor(instances, filter) {
+    let spec = this.props.pod.getSpec();
+
+    return instances.getItems().map(function (instance) {
+      let resourcesSummary = {
+        cpus: 0, mem: 0, disk: 0, gpus: 0
+      };
+
+      let children = instance.getContainers()
+        .filter(function (container) {
+          return PodUtil.isContainerMatchingText(container, filter.text);
+        })
+        .map(function (container) {
+          let containerSpec = spec.getContainerSpec(container.name);
+          Object.keys(containerSpec.resources).forEach(function (key) {
+            resourcesSummary[key] = containerSpec.resources[key];
+          });
+
+          return {
+            id: container.getId(),
+            name: container.getName(),
+            status: container.getContainerStatus(),
+            address: container.endpoints.reduce(function (components, ep) {
+              components.push(
+                <a className="text-muted"
+                  href={
+                    'http://' +
+                    instance.getAgentAddress() + ':' +
+                    ep.allocatedHostPort
+                  }
+                  target="_blank"
+                  title="Open in a new window">
+                  {':' + ep.allocatedHostPort}
+                </a>
+              );
+
+              return components;
+            }, []),
+            cpus: containerSpec.resources.cpus,
+            mem: containerSpec.resources.mem,
+            updated: container.getLastUpdated(),
+            version: ''
+          };
+        });
+
+      return {
+        id: instance.getId(),
+        name: instance.getName(),
+        address: instance.getAgentAddress(),
+        cpus: resourcesSummary.cpus,
+        mem: resourcesSummary.mem,
+        updated: instance.getLastUpdated(),
+        status: instance.getInstanceStatus(),
+        version: spec.getVersion(),
+        children
+      };
+    });
+
+  }
+
+  getInstanceFilterStatus(instance) {
```

Comment:
@mlunoe this is a view specific helper and we shouldn't add those to the structs, but it could be useful to move it to a util.